src/ucioption.h

```diff
@@ -72,10 +56,38 @@ class Option {
     friend std::ostream& operator<<(std::ostream&, const OptionsMap&);
 
    private:
+    friend class OptionsMap;
+    friend class Engine;
+    friend class Tune;
+
+    void listen(OnUpdate&&);
+    void operator<<(const Option&);
+
     std::string defaultValue, currentValue, type;
     int         min, max;
     size_t      idx;
     OnChange    on_change;
+    OnUpdate    listener = nullptr;
```

Comment:
Confusing type names. perhaps `OnChangeListener` would be better, if we keep it at all